.github/workflows/CI.yml

```diff
@@ -1,10 +1,19 @@
 name: CI
+
 on:
-  - push
-  - pull_request
+  push:
+    branches: [master]
+  pull_request:
+
+concurrency:
+  # Skip intermediate builds: always.
+  # Cancel intermediate builds: only if it is a pull request build.
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ startsWith(github.ref, 'refs/pull/') }}
```

Comment:
Ah, I hadn't seen this done before. I assume you've seen it in action and it works as documented.

Reply:
Yeah, it's used all over the place (e.g. ChainRulesCore: https://github.com/JuliaDiff/ChainRulesCore.jl/blob/3e11e627b5a10a3149902b2c538be63c61ecafbc/.github/workflows/CI.yml#L10-L14) and part of PkgTemplates (https://github.com/invenia/PkgTemplates.jl/blob/1cb56bf90326d47c402af51537c9c669e9b08a24/templates/github/workflows/CI.yml#L10-L14).